artifacts/artifact.py

```diff
@@ -50,35 +50,8 @@ def AppendSource(self, type_indicator, attributes):
     if not type_indicator:
       raise errors.FormatError(u'Missing type indicator.')
 
-    source_type_class = None
-    if type_indicator == definitions.TYPE_INDICATOR_ARTIFACT_GROUP:
-      source_type_class = source_type.ArtifactGroupSourceType
-
-    elif type_indicator == definitions.TYPE_INDICATOR_COMMAND:
-      source_type_class = source_type.CommandSourceType
-
-    elif type_indicator == definitions.TYPE_INDICATOR_COMMAND:
-      source_type_class = source_type.CommandCollectorDefinition
-
-    elif type_indicator == definitions.TYPE_INDICATOR_DIRECTORY:
-      source_type_class = source_type.DirectorySourceType
-
-    elif type_indicator == definitions.TYPE_INDICATOR_FILE:
-      source_type_class = source_type.FileSourceType
-
-    elif type_indicator == definitions.TYPE_INDICATOR_PATH:
-      source_type_class = source_type.PathSourceType
-
-    elif type_indicator == definitions.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY:
-      source_type_class = source_type.WindowsRegistryKeySourceType
-
-    elif type_indicator == definitions.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE:
-      source_type_class = source_type.WindowsRegistryValueSourceType
-
-    elif type_indicator == definitions.TYPE_INDICATOR_WMI_QUERY:
-      source_type_class = source_type.WMIQuerySourceType
-
-    else:
+    source_type_class = source_type.SOURCE_TYPES.get(type_indicator)
+    if source_type_class is None:
```

Comment:
Using is None after a .get() is idiomatic python. Calling a boolean operator here is not the most correct because it calls any overriden __nonzero__(), __len__() etc. Its also faster to use the "is None" idiom here.